tests/telemetry/test_telemetry_poll.py

```diff
@@ -239,3 +239,64 @@ def test_poll_mode_default_route_supervisor(duthosts, enum_rand_one_per_hwsku_ho
     update_responses_match = re.findall("json_ietf_val", result)
     pytest_assert(len(update_responses_match) == 5, "Missing update responses")
     modify_fake_appdb_table(duthost, False, 1)  # Remove added table
+
+
+@pytest.mark.parametrize('setup_streaming_telemetry', [False], indirect=True)
+def test_poll_mode_srv6_sid_counters(duthosts, enum_rand_one_per_hwsku_hostname, ptfhost,
+                                     setup_streaming_telemetry, gnxi_path):
+    """
+    Test poll mode from COUNTERS_DB and query SRv6 MY_SID counters:
+    First, query when the data does not exist,ensure no errors and present data
+    Second, add data and then test query again ensuring data comes.
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    namespace = ""
+    if duthost.is_multi_asic:
+        namespace = "asic0"
+    logger.info('Start telemetry poll mode testing')
+    cmd = generate_client_cli(duthost=duthost, gnxi_path=gnxi_path, method=METHOD_SUBSCRIBE,
+                              subscribe_mode=SUBSCRIBE_MODE_POLL, polling_interval=2,
+                              xpath="\"COUNTERS/SID:*\"",  # noqa: W605
+                              target="COUNTERS_DB", max_sync_count=-1, update_count=5, timeout=30, namespace=namespace)
+
+    ptf_result = ptfhost.shell(cmd)
+    pytest_assert(ptf_result['rc'] == 0, "ptf cmd command {} failed".format(cmd))
+    show_gnmi_out = ptf_result['stdout']
+    logger.info("GNMI Server output")
+    logger.info(show_gnmi_out)
+    result = str(show_gnmi_out)
+    update_responses_match = re.findall("json_ietf_val", result)
+    pytest_assert(len(update_responses_match) > 0, "Incorrect update responses")
+
+    if namespace != "":
+        SONIC_DB_CLI = f"sonic-db-cli -n {namespace}"
+    else:
+        SONIC_DB_CLI = "sonic-db-cli"
+    # Now generate some SRv6 SID counter values by adding mock data
+    duthost.shell(f"{SONIC_DB_CLI} COUNTERS_DB HSET \"COUNTERS:oid:0x11110000001eb3\" SAI_COUNTER_STAT_PACKETS 10 \
```

Comment:
wondering if we also add another test that is specific to 5640/7060x6 SKUs such that we do `sudo counterpoll srv6 enable` instead of manually setting, would that work? If the HardwareSKU is not that, then we skip that test.